### ISSUE TITLE
Test do not run

### ISSUE THREAD

**Issue by siuying** [rayh/xcoder#38]
I'm using Xcoder to automate tests. I just create a new Xcode project with Test from template, and use Xcoder to run test, it return no result.

Look into the debug logs, following command line is executed:

```
xcodebuild -sdk iphonesimulator -project "/Users/siuying/Documents/workspace/testflight-uploader/spec/fixtures/projects/SampleApp2/SampleApp2.xcodeproj" -target "SampleApp2Tests" -configuration "Debug" OBJROOT="/Users/siuying/Documents/workspace/testflight-uploader/spec/fixtures/projects/SampleApp2/build/" SYMROOT="/Users/siuying/Documents/workspace/testflight-uploader/spec/fixtures/projects/SampleApp2/build/" TEST_AFTER_BUILD=YES
```

Try to run it on command line myself, i see following messages:

```
/bin/sh -c /Users/siuying/Documents/workspace/testflight-uploader/spec/fixtures/projects/SampleApp2/build/SampleApp2.build/Debug-iphonesimulator/SampleApp2Tests.build/Script-54F4F50E1536C2F000287878.sh
/Developer/Platforms/iPhoneSimulator.platform/Developer/Tools/Tools/RunPlatformUnitTests:95: warning: Skipping tests; the iPhoneSimulator platform does not currently support application-hosted tests (TEST_HOST set).
```

I wonder if the change at #26 related to this. In my case, append TEST_HOST='' in the parameters fix the error, the test will run as expected.

I'm using XCode 4.3.2 4E2002

**Comment by rayh** [comment]
Damn, it seems that flag is needed when running against iphonesimulator.  I'll take a look at doing something that adds it only if its testing against the simulator sdk... this should hopefully keep it out of the way when running OSX tests.

**Comment by rayh** [comment]
Ok, there is a bigger issue here.  Do you mean to run Application tests (i.e. ones that use UIKit?).  Logic tests should run fine without TEST_HOST='' and specifying it is causing issue #43

Could you provide a spec (maybe in integration/builder_spec.rb) that shows this failure when I remove TEST_HOST='' ?

**Comment by rayh** [comment]
Sorry, I lied, it is failing when I remove it. Digging deeper...

**Comment by siuying** [comment]
Still need a spec for this?

Just tried. The problem i found is running test has no effect, even it should fail, it just complete without error. The expect case (at 0.1.11) is when test fail, the process exit.

**Comment by rayh** [comment]
I think it's ok, I have something reproducible now... Seems to be
because Xcode thinks its an application test, but I can't work out how
to set it to be a Logic test

On 17 April 2012 17:03, siuying
reply@reply.github.com
wrote:

> Still need a spec for this?
> 
> Just tried. The problem i found is running test has no effect, even it should fail, it just pass.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/rayh/xcoder/issues/38#issuecomment-5169452

## 

## Ray Hilton

a: 13/243 Collins Street, Melbourne VIC 3000
m: 0430 484 708
w: http://rayh.com.au

**Comment by blakewatters** [comment]
I believe that the only difference between an iOS Unit Test and an Application Tests is the presence of the BUNDLE_LOADER and TEST_HOST variables at build time, which instructs the system to inject the test bundle into a host application that it can resolve symbols dynamically to test the host application and access the UIApplication reference. I think that the right thing to do here may be to just fix the host project by clearing those build variables on the test target. From Xcoder's perspective, I think its impossible to infer whether you are trying to fire up logic or application tests at build time.

**Comment by rayh** [comment]
Ok, a suggestion is to set TEST_HOST='' in your project configuration for the logic tests rather than override it from within xcoder.  Setting TEST_HOST is valid for running application tests, which are sort of not supported by xcodebuild.

Can you give that a go and let me know?

**Comment by rayh** [comment]
See #43, but this is fixed in 283027d

**Comment by siuying** [comment]
@rayh I tried to run the test, but the TestProject would not build with many missing dependencies. 

Do you mean that for every project we need tests, we have to add TEST_HOST in XCode in order we automate the test in xcoder? Ideally we want automate without changing the project manually.

Would it be okay if we make the TEST_HOST optional? such as:

```
builder.test(:test_host => false)
```

or maybe 

```
builder.test(:application_test => true)
```

at least it would run in both case.
